### PR TITLE
feat: support ldname attribute

### LINKED ISSUE #1326: Support ldName attribute
As a user of OpenSCD I want to use functional naming in order to create vendor independent SCD files consisting virtual IEDs without referring to IED name+LDinst attributes.

### PR DESCRIPTION
**Pull request by Stef3st**
closes #1326

### CODE REVIEW

**Review by juancho0202 — CHANGES_REQUESTED** [review]
Two very minor comments, it's pretty much all done. Nice!

File: packages/open-scd/test/unit/wizards/ldevice.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    it('ConfLdName should not be present and therefore ldName should be readonly', async function () {
+      expect(services.querySelector('ConfLdName')).to.not.exist;
+      expect(<WizardTextField>inputs[0]).to.have.attribute('readonly');
+    });
+
+    it('looks like the latest snapshot', async () => {
+      await expect(element.wizardUI.dialog).dom.to.equalSnapshot();
+    });
+
+    it('ConfLdName should be present in IED1 and therefore ldName should be readonly', async function () {
```
[the remark this answers is not in the document]
Comment (Stef3st) [reply packages/open-scd/test/unit/wizards/ldevice.test.ts:61]: Oooh nice catch! Thats what you get for copy/pasting ☹️

File: packages/open-scd/test/unit/wizards/ldevice.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    ).to.be.equal('');
+  });
+  it('contains a wizard-textfield with an empty "ldName" value', async () => {
+    expect(
+      (<WizardTextField[]>inputs).find(textField => textField.label == 'ldName')
+        ?.value
+    ).to.be.equal('');
+  });
+
+  describe('Allowing ldName editing', () => {
```
[the remark this answers is not in the document]
Comment (Stef3st) [reply packages/open-scd/test/unit/wizards/ldevice.test.ts:51]: Makes more sense as both cases are tested here. Will change it!

**Comment by DavoodSooran** [comment]
I tested the updated Support LDname feature and it's great that we can now enable/disable the LDname section. However, there's a functionality improvement needed. When enabling LDname and Description, it should require user input before saving. Currently, it's possible to save without entering information, resulting in empty sections in the SCL file. I suggest implementing an error prompt for empty inputs when users try to save after enabling these sections.

image

image

**Comment by Stef3st** [comment]
@DavoodSooran we then would be tampering with the entire wizarding functionality. If you don't want the elements to be in the SCL then you should disable the fields with the null switch.

**Review by juancho0202 — APPROVED** [review]

**Pull request merged**